walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,68 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> eventRanges = sortedRangesInQuestion(events, request.getAttendees());
+    List<TimeRange> possibleRanges = new ArrayList<TimeRange>();
+    int startingPointer = 0;
+    int endingPointer = 1;
+    System.out.println(eventRanges.size());
+    while(endingPointer < eventRanges.size()) {
+      if(eventRanges.get(startingPointer).contains(eventRanges.get(endingPointer))) {
+        endingPointer++;
+      }
+      else if(eventRanges.get(startingPointer).overlaps(eventRanges.get(endingPointer))) {
+        startingPointer = endingPointer;
+        endingPointer++;
+      }
+      else {
+        TimeRange between = TimeRange.fromStartEnd(eventRanges.get(startingPointer).end(),
+            eventRanges.get(endingPointer).start(), false);
+        if(between.duration() >= request.getDuration()){
+          possibleRanges.add(between);
+        }
+        startingPointer++;
+        endingPointer++;
```

Comment:
endingPointer seems to always increment, so you could move all the endPointer++ statements outside of the if-statements?

Additionally, if that's the case, you could also change this while loop to a for loop, since you have endingPointer starting as 1, always incrementing by 1, and being smaller than eventRanges.size().